src/main/java/core/basesyntax/strategy/SupplyOperationHandler.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import java.util.Optional;
+
+public class SupplyOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        if (fruitTransaction == null) {
+            throw new IllegalArgumentException("FruitTransaction cannot be null");
+        }
+        String fruit = fruitTransaction.getFruit();
+        if (fruit == null || fruit.isEmpty()) {
+            throw new IllegalArgumentException("Fruit cannot be null or empty");
+        }
+        int suppliedAmount = fruitTransaction.getQuantity();
+        if (suppliedAmount <= 0) {
+            throw new IllegalArgumentException("Quantity must be greater than zero");
+        }
+
+        Optional<Integer> currentBalance = Optional.ofNullable(Storage.fruitStorage
```

Comment:
It's recommended to use data structures from the Collection framework to represent fruit storage in a more appropriate way. The `Storage` class should be encapsulated and accessed through methods rather than directly.